libraries/chain/dbs_reward.cpp

```diff
@@ -10,6 +10,18 @@ dbs_reward::dbs_reward(database& db)
 {
 }
 
+const reward_pool_object& dbs_reward::create_pool(const asset& initial_supply)
```

Comment:
Add more checks: 
Add token type to object reward_pool_object (default initialize) and check symbol in service. Budget uses SCORUM_SYMBOL.
Add SCORUM_BLOCKS_PER_DAY , SCORUM_GUARANTED_REWARD_SUPPLY_PERIOD_IN_DAYS checks
to prevent division by zero

Reply:
I've created a separate task for this. https://scorum.atlassian.net/browse/BLOC-156